src/Http/Runner.php

```diff
@@ -72,9 +65,10 @@ public function run(
      */
     public function handle(ServerRequestInterface $request): ResponseInterface
     {
-        $middleware = $this->queue->get($this->index++);
+        if ($this->queue->valid()) {
+            $middleware = $this->queue->current();
+            $this->queue->next();
 
```

Comment:
Operating an iterator like this is a bit clunky would `yield` be usable here?

Reply:
Nope it isn't.

Runners in other libs I check also traverse the queue manually. `Relay` use `current()` and `next()` functions on it's queue which is an array. `zend-stratigility`'s queue is an `SplQueue` and it's runner uses `dequeue()` method to get next middleware.

Reply:
Ok. It was worth a shot.